orchagent/nexthopkey.h

```diff
@@ -98,6 +117,73 @@ struct NextHopKey
     {
         return (ip_address.getV4Addr() == 0);
     }
+
+    bool isMplsNextHop() const
+    {
+        return (label_stack.getSize() > 0);
+    }
+
+    std::string parse_mpls(const std::string& str)
+    {
+        std::string ip_str;
+        auto keys = tokenize(str, LABELSTACK_DELIMITER);
+        if (keys.size() == 1)
+        {
+            // No MPLS info to parse
+            ip_str = str;
+        }
+        else if (keys.size() == 3)
+        {
+            // Expected MPLS format = "<outsegtype>+<labelstack>+<non-mpls-str>"
+            // key[0] = <outsegtype> = "swap" | "pop"
+            // key[1] = <labelstack> = "<label0>/<label1>/../<labelN>"
+            // key[2] = <non-mpls-str> = returned to caller and not parsed here
+            // Example = "push+10100/10101+10.0.0.3@Ethernet4"
+            if (keys[0] == "swap")
+            {
+                outseg_type = SAI_OUTSEG_TYPE_SWAP;
+            }
+            else if (keys[0] == "push")
+            {
+                outseg_type = SAI_OUTSEG_TYPE_PUSH;
+            }
+            else
+            {
+                // Malformed string
+                std::string err = "Error converting " + str + " to MPLS NextHop";
+                throw std::invalid_argument(err);
+            }
+            label_stack = LabelStack(keys[1]);
+            ip_str = keys[2];
+        }
+        else
+        {
+            // Malformed string
+            std::string err = "Error converting " + str + " to MPLS NextHop";
+            throw std::invalid_argument(err);
+        }
+        return ip_str;
+    }
+
+    std::string format_mpls() const
+    {
+        std::string str;
+        if (!label_stack.empty())
+        {
+            if (outseg_type == SAI_OUTSEG_TYPE_SWAP)
+            {
+                str += "swap";
+                str += LABELSTACK_DELIMITER;
+            }
+            else if (outseg_type == SAI_OUTSEG_TYPE_PUSH)
+            {
+                str += "push";
+                str += LABELSTACK_DELIMITER;
+            }
+            str += label_stack.to_string() + LABELSTACK_DELIMITER;
```

Comment:
is there a need for "LABELSTACK_DELIMITER" at the end?

Reply:
Yes, this is will delimit the MPLS component from the rest (non-MPLS) of the next hop string.